diff --git a/core/modules/views/src/EntityViewsData.php b/core/modules/views/src/EntityViewsData.php
index ce2ac1519c..675a042a6b 100644
--- a/core/modules/views/src/EntityViewsData.php
+++ b/core/modules/views/src/EntityViewsData.php
@@ -347,6 +347,12 @@ public function getViewsData() {
             ['field' => 'deleted', 'value' => 0, 'numeric' => TRUE],
           ],
         ];
+        if ($data_table && $field_definition->isTranslatable()) {
+          $data[$table]['table']['join'][$views_base_table]['extra'][] = [
+            'field' => $entity_keys['langcode'],
+            'left_field' => $entity_keys['langcode'],
+          ];
+        }
 
         if ($revisionable) {
           $revision_table = $table_mapping->getDedicatedRevisionTableName($field_definition->getFieldStorageDefinition());
@@ -360,6 +366,12 @@ public function getViewsData() {
               ['field' => 'deleted', 'value' => 0, 'numeric' => TRUE],
             ],
           ];
+          if ($revision_data_table && $field_definition->isTranslatable()) {
+            $data[$revision_table]['table']['join'][$views_revision_base_table]['extra'][] = [
+              'field' => $entity_keys['langcode'],
+              'left_field' => $entity_keys['langcode'],
+            ];
+          }
         }
       }
     }
diff --git a/core/modules/views/tests/src/Kernel/Entity/EntityViewsDataTest.php b/core/modules/views/tests/src/Kernel/Entity/EntityViewsDataTest.php
index 6fc3c673d1..7c63d92c87 100644
--- a/core/modules/views/tests/src/Kernel/Entity/EntityViewsDataTest.php
+++ b/core/modules/views/tests/src/Kernel/Entity/EntityViewsDataTest.php
@@ -513,6 +513,10 @@ public function testDataTableFields() {
           'value' => 0,
           'numeric' => TRUE,
         ],
+        [
+          'field' => 'langcode',
+          'left_field' => 'langcode',
+        ],
       ],
     ], $data['entity_test_mul__string']['table']['join']['entity_test_mul_property_data']);
   }
@@ -644,6 +648,10 @@ public function testRevisionTableFields() {
           'value' => 0,
           'numeric' => TRUE,
         ],
+        [
+          'field' => 'langcode',
+          'left_field' => 'langcode',
+        ],
       ],
     ], $data['entity_test_mulrev__string']['table']['join']['entity_test_mulrev_property_data']);
 
@@ -658,6 +666,10 @@ public function testRevisionTableFields() {
           'value' => 0,
           'numeric' => TRUE,
         ],
+        [
+          'field' => 'langcode',
+          'left_field' => 'langcode',
+        ],
       ],
     ], $data['entity_test_mulrev_revision__string']['table']['join']['entity_test_mulrev_property_revision']);
   }
diff --git a/core/modules/views/views.views.inc b/core/modules/views/views.views.inc
index e24be15b90..fb9fba45c9 100644
--- a/core/modules/views/views.views.inc
+++ b/core/modules/views/views.views.inc
@@ -837,6 +837,12 @@ function core_field_views_data(FieldStorageConfigInterface $field_storage) {
           ],
         ],
       ];
+      if ($target_base_table && $field_storage->isTranslatable()) {
+        $data[$target_base_table][$pseudo_field_name]['relationship']['join_extra'][] = [
+          'field' => 'langcode',
+          'left_field' => 'langcode',
+        ];
+      }
     }
   }
 
